presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1172,21 +1173,27 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
     @Override
     public Optional<ConnectorNewTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        HivePartitionResult hivePartitionResult = partitionManager.getPartitions(metastore, tableHandle, Constraint.alwaysTrue());
-        if (!hivePartitionResult.getBucketHandle().isPresent()) {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
+        SchemaTableName tableName = hiveTableHandle.getSchemaTableName();
+        Optional<Table> table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName());
```

Comment:
You can write this as
```java
Table table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName())
        .orElseThrow(() -> new TableNotFoundException(tableName));
```
(we should use this trick in more places, though most of this code pre-dates Java 8)